modules/statistics/AvgRTPStatsReporter.js

```diff
@@ -75,6 +75,205 @@ class AverageStatReport {
+ */
+class ConnectionAvgStats {
+    /**
+     * Creates new <tt>ConnectionAvgStats</tt>
+     * @param {JitsiConference} conference
+     * @param {boolean} isP2P
+     * @param {number} n the number of samples, before arithmetic mean is to be
+     * calculated and values submitted to the analytics module.
+     */
+    constructor(conference, isP2P, n) {
+        /**
+         * Is this instance for JVB or P2P connection ?
+         * @type {boolean}
+         */
+        this.isP2P = isP2P;
+
+        /**
+         * How many samples are to be included in arithmetic mean calculation.
+         * @type {number}
+         * @private
+         */
+        this._n = n;
+
+        /**
+         * The current sample index. Starts from 0 and goes up to {@link _n})
+         * when analytics report will be submitted.
+         * @type {number}
+         * @private
+         */
+        this._sampleIdx = 0;
+
+        /**
+         * Average round trip time reported by the ICE candidate pair.
+         * @type {AverageStatReport}
+         */
+        this._avgRTT = new AverageStatReport('stat.avg.rtt');
+
+        /**
+         * Map stores average RTT to the JVB reported by remote participants.
+         * Mapped per participant id {@link JitsiParticipant.getId}.
+         *
+         * This is used only when {@link ConnectionAvgStats.isP2P} equals to
+         * <tt>false</tt>.
+         *
+         * @type {Map<string,AverageStatReport>}
+         * @private
+         */
+        this._avgRemoteRTTMap = new Map();
+
+        /**
+         * The conference for which stats will be collected and reported.
+         * @type {JitsiConference}
+         * @private
+         */
+        this._conference = conference;
+
+        this._onConnectionStats = (tpc, stats) => {
+            if (this.isP2P === tpc.isP2P) {
+                this._calculateAvgStats(stats);
+            }
+        };
+        conference.statistics.addConnectionStatsListener(
+            this._onConnectionStats);
+
+        if (!this.isP2P) {
+            this._onUserLeft = id => this._avgRemoteRTTMap.delete(id);
+            conference.on(ConferenceEvents.USER_LEFT, this._onUserLeft);
+
+            this._onRemoteStatsUpdated
+                = (id, data) => this._processRemoteStats(id, data);
+            conference.on(
+                ConnectionQualityEvents.REMOTE_STATS_UPDATED,
+                this._onRemoteStatsUpdated);
+        }
+    }
+
+    /**
+     * Processes next batch of stats.
+     * @param {go figure} data
+     * @private
+     */
+    _calculateAvgStats(data) {
+        if (!data) {
+            logger.error('No stats');
+
+            return;
+        }
+
+        if (RTCBrowserType.supportsRTTStatistics()) {
+            if (data.transport && data.transport.length) {
+                this._avgRTT.addNext(data.transport[0].rtt);
+            } else {
+                this._avgRTT.reset();
```

Comment:
I would expect that lack of transport will mean that transport changed or something is wrong (more likely the latter), so I thought it should be reset to make sense for the new transport. Each stat carries it's own sum and count. But maybe it doesn't really make sense. I'll remove -
 it's only causing unnecessary confusion.